src/app/client/src/app/modules/dashboard/components/report/report.component.ts

```diff
@@ -575,12 +582,10 @@ export class ReportComponent implements OnInit {
   }
 
   getChartData(chart){
-   let chartInfo = this.chartsReportData.charts.filter(data=>{
-        if(data.chartConfig){
-          if(data.chartConfig.id == chart.chartConfig.id){
-            return data;
-          }
-        }
+    let chartInfo = this.chartsReportData.charts.filter(data=>{
```

Comment:
This isn't the right & efficient way of filtering out charts. Since you need to find a chart inside the array use 
Array.prototype.find() method

Reply:
resolved